### DISCUSSION TITLE
How to compose 3 by 3 image chunk in parallel

### DISCUSSION THREAD

**Discussion by spd789562** [image-rs/image#2174] [Q&A]
I have a encoded image which parallel decoded into many of 3 by 3 chunk of image, but it will need compose them in single thread later something like.
```rust
let image_buffer_chunks: Vec<ImageBuffer<Rgba<u8>, Vec<u8>>> = raw.par_chunks(9).map(|r| {
  // do some stuff
  image::ImageBuffer::from_par_fn(3, 3, |x, y| { color_table[(y * 3 + x) as usize] })
}).collect();

// put buffers in
let mut img = DynamicImage::new_rgba8(width, height);
let grid_row_count = width / 3;
for (i, chunk) in image_buffer_chunks.iter().enumerate() {
  let x = (i as u32 % grid_row_count) * 3;
  let y = (i as u32 / grid_row_count) * 3;
  img.copy_from(chunk, x, y).unwrap();
}

Ok(img)
```
Is there anyway to doing this in single iter or I'm missing something? current approach is not even quicker then doing whole process in single thread.

Maybe it in rayon's iter, but i can't find it.

**Comment by fintelia** [comment]
You could try something like:

```rust
let img = DynamicImage::ImageRgba8(ImageBuffer::from_par_fn(width, height, |x,y| {
    image_buffer_chunks[(x/3) + (y/3) * grid_row_count].get_pixel(x%3, y%3)
}));
```

**Reply by spd789562** [reply]
Nice, it actually faster, thank you, but then I'm find out the first part of `image::ImageBuffer::from_par_fn` is unnecessary cuz it really small. 

So I'm changed to simple loop, and it finally quicker then single thread version(put_pixel one by one) of whole proces 🚀 . 

But I'm still think is there a way to do this in one parallel loop thou. if I can get every part of 3*3 images somehow and doing parallel decode
